ngraph/core/src/op/interpolate.cpp

```diff
@@ -183,12 +195,15 @@ void op::v4::Interpolate::infer_using_scales(PartialShape& output_shape,
     size_t i = 0;
     for (auto axis : axes)
     {
-        if (padded_input_shape[axis].is_static())
-        {
-            float padded_len = static_cast<float>(padded_input_shape[axis].get_length());
-            int64_t new_dim = static_cast<int64_t>(padded_len * (scales[i] + epsilon));
-            output_shape[axis] = Dimension(new_dim);
-        }
+        const auto& current_dim = padded_input_shape[axis];
+        float multiplier = scales[i] + epsilon;
+
+        int64_t new_lower_bound =
+            multiply_bound_and_scale(current_dim.get_min_length(), multiplier);
+        int64_t new_upper_bound =
+            multiply_bound_and_scale(current_dim.get_max_length(), multiplier);
```

Comment:
Don't we have the same logic for Interval class, i.e. multiplication of Interval by a scalar? I think we need to update `Interval` class with this method if not. 

@jane-intel, what do you think of the idea about updating a class `Interval` wit a method of multiplication an interval by a scalar?

Reply:
No, there are no same logic in the `Interval` class.

Reply:
That is why I propose to add it

Reply:
But there are no other operation using a multiplication of `Interval` with scalar. Moreover, we can get an overloading conflict for the multiplication operation, because there is an implicit conversion from `int64_t` into `Interval`, and an implicit conversion from `int64_t` into `float`. We have an implicit conversion from `int64_t` into `Interval`, because the class `Interval` has a non-explicit constructor `Interval(int64_t)`.